llvm/lib/Target/AMDGPU/SIISelLowering.cpp

```diff
@@ -6822,6 +6822,81 @@ static unsigned getExtOpcodeForPromotedOp(SDValue Op) {
   }
 }
 
+SDValue SITargetLowering::combineAnd(SDValue Op,
```

Comment:
You should not be implementing any new combine in this patch. The point of this is to enable the existing optimizations. If it's worth adding something else, it would be a separate PR

Reply:
Okay, but I think hasAndNot can use And operation when meet this scenario: `(and LHS, (or Y, ~Z))` and X86 used this way, so I try to implement it. :)

Reply:
I don't know why x86 has that in the backend but it belongs in a generic combine

Reply:
Okay, I have added this feature to promoteUniformOpToI32 and removed it. :)